projects/scripts/fmc_constr_generator.tcl

```diff
@@ -0,0 +1,276 @@
+proc gen_fmc_constr {{fmc_index1 fmc} {fmc_index2 {}}} {
+  
+  set project_name [get_property NAME [current_project]]
+  set carrier [string replace $project_name 0 [string first "_" $project_name] ""]
+  set eval_board [string replace $project_name [string first "_" $project_name] 30 ""]
+  
+  variable col0_max 0
+  variable col1_max 0
+  variable col2_max 0
+  variable col3_max 0
+  variable io_standard_max 0
+  variable temp_pin_e 0
+  variable rx_bank
+  variable tx_bank
+  
+  set carrier_path [glob ../../common/$carrier/$carrier\_$fmc_index1*.txt]
+  set carrier_file [open $carrier_path r]
+  set carrier_data [read $carrier_file]
+  close $carrier_file
+  set line_c [join $carrier_data " "]
+  
+  set constr_file [open "fmc_constr.xdc" w+]
+  
+  if {[string length $fmc_index2] == 0} {
+    set eval_board_path [glob ../common/$eval_board\_fmc*.txt]
```

Comment:
Keep in mind that the project name is not always the same with the _bd.tcl
I would go with set eval_board_path [glob ../common/*_fmc*.txt]